src/marqo/s2_inference/hf_utils.py

```diff
@@ -16,15 +24,95 @@ def __init__(self, *args, **kwargs) -> None:
 
         if self.max_seq_length is None:
             self.max_seq_length = 128
+        self.model_properties = kwargs.get("model_properties", dict())
+        self.model_name = self.model_properties.get("name", None)
+
+        model_auth = kwargs.get(InferenceParams.model_auth, None)
+        if model_auth is not None:
+            self.model_auth = model_auth
+        else:
+            self.model_auth = None
 
     def load(self) -> None:
 
-        self.model = AutoModelForSentenceEmbedding(self.model_name).to(self.device)
-        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+        model_location_presence = ModelProperties.model_location in self.model_properties
+        path = self.model_properties.get("localpath", None) or self.model_properties.get("url", None)
+        # HF models can be loaded from 3 entries: path (url or localpath), model_name, or model_location
+        if (path is not None) + (self.model_name is not None) + (model_location_presence is True) != 1:
+            raise InvalidModelPropertiesError("Exactly one of `url`, `localpath` or `model_location`, `name` can be specified"
```

Comment:
Is this exactly what the issue would be? 
It can be either 'localpath' or 'url' but only one of (localpath/url), model_name or ModelProperties.model_location

Reply:
updated